gensim/models/fasttext_inner.pyx

```diff
@@ -647,111 +657,164 @@ def train_batch_sg(model, sentences, alpha, _work, _l1):
         int num_words = 0
         int num_sentences = 0
 
-    init_ft_config(&c, model, alpha, _work, _l1)
+    init_ft_config(&c, model, alpha, _work, _neu1)
 
-    num_words, num_sentences = populate_ft_config(&c, model.wv.vocab, model.wv.buckets_word, sentences)
+    num_words, num_sentences = populate_ft_config(&c, model.wv, model.wv.buckets_word, sentences)
 
     # precompute "reduced window" offsets in a single randint() call
     for i, randint in enumerate(model.random.randint(0, c.window, num_words)):
         c.reduced_windows[i] = randint
 
+    # release GIL & train on all sentences in the batch
     with nogil:
-        fasttext_train_any(&c, num_sentences, 1)
+        fasttext_train_any(&c, num_sentences, 0)
```

Comment:
Why `0`?

Reply:
Good catch! The refactoring should have moved this mode parameter into the `FastTextConfig` object, and then consulted it there.

Reply:
Alright. Can you fix it before the merge?

I believe this is the only bug, the rest are design decisions we can address in subsequent PRs. But let's not commit bugs.

Reply:
Fixed in 5c7eb1c.